### PR TITLE
[Platform] Enable L10n linter on new betas and releases

### PR DESCRIPTION
**Pull request by AliSoftware**
### Fix

This PR enables the linting of localizations by calling the action (previously introduced by wordpress-mobile/release-toolkit#174).

References: paaHJt-D6-p2 & paNNhX-am-p2 (Step 1)

### Test

* Run the `new_beta_release` or `finalize_release` lanes and check that the linter got run.
* Alternatively, to test without doing a new beta or release, just run `bundle exec fastlane run ios_lint_localizations input_dir:Simplenote/` (ideally after having run `Scripts/update-translations.rb` to pick the latest updates, tho I've already ran that one in this PR)

<details><summary>Expected output: no violations</summary>

```$ fl run ios_lint_localizations input_dir:Simplenote/
[✔] 🚀 
/Users/olivier/Documents/Dev/simplenote-ios/vendor/bundle/ruby/2.6.0/gems/activesupport-4.2.11.3/lib/active_support/core_ext/object/duplicable.rb:111: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
+-----------------------------------+---------+---------------------------------------------------+
|                                          Used plugins                                           |
+-----------------------------------+---------+---------------------------------------------------+
| Plugin                            | Version | Action                                            |
+-----------------------------------+---------+---------------------------------------------------+
| fastlane-plugin-wpmreleasetoolkit | 0.10.0  | configure_download configure_setup                |
|                                   |         | configure_apply configure_update                  |
|                                   |         | configure_add_files_to_copy configure_validate    |
|                                   |         | ios_build_prechecks ios_betabuild_prechecks       |
|                                   |         | ios_build_preflight ios_localize_project          |
|                                   |         | ios_update_release_notes                          |
|                                   |         | add_devices_to_provisioning_profiles              |
|                                   |         | ios_current_branch_is_hotfix                      |
|                                   |         | add_development_certificates_to_provisioning_pro  |
|                                   |         | files ios_merge_translators_strings               |
|                                   |         | ios_hotifx_prechecks ios_tag_build ios_final_tag  |
|                                   |         | ios_finalize_prechecks ios_bump_version_hotfix    |
|                                   |         | ios_check_beta_deps ios_validate_ci_build         |
|                                   |         | ios_get_app_version ios_lint_localizations        |
|                                   |         | ios_codefreeze_prechecks ios_update_metadata      |
|                                   |         | ios_clear_intermediate_tags                       |
|                                   |         | ios_bump_version_beta                             |
|                                   |         | ios_completecodefreeze_prechecks                  |
|                                   |         | ios_update_metadata_source                        |
|                                   |         | ios_bump_version_release ios_get_store_app_sizes  |
|                                   |         | extract_release_notes_for_version                 |
|                                   |         | gp_downloadmetadata create_release setfrozentag   |
|                                   |         | close_milestone promo_screenshots                 |
|                                   |         | removebranchprotection gp_update_metadata_source  |
|                                   |         | get_prs_list create_new_milestone                 |
|                                   |         | setbranchprotection                               |
|                                   |         | android_completecodefreeze_prechecks              |
|                                   |         | android_betabuild_prechecks                       |
|                                   |         | android_bump_version_release                      |
|                                   |         | android_current_branch_is_hotfix                  |
|                                   |         | android_build_prechecks                           |
|                                   |         | android_bump_version_final_release                |
|                                   |         | android_build_preflight                           |
|                                   |         | android_get_release_version                       |
|                                   |         | android_codefreeze_prechecks android_tag_build    |
|                                   |         | android_create_xml_release_notes                  |
|                                   |         | android_hotifx_prechecks an_localize_libs         |
|                                   |         | android_bump_version_hotfix                       |
|                                   |         | android_update_release_notes                      |
|                                   |         | android_update_metadata an_validate_lib_strings   |
|                                   |         | android_get_alpha_version                         |
|                                   |         | android_get_app_version                           |
|                                   |         | android_merge_translators_strings                 |
|                                   |         | android_finalize_prechecks                        |
|                                   |         | an_update_metadata_source                         |
|                                   |         | android_bump_version_beta                         |
| fastlane-plugin-sentry            | 1.6.0   | sentry_upload_dsym sentry_set_commits             |
|                                   |         | sentry_upload_sourcemap sentry_finalize_release   |
|                                   |         | sentry_upload_file sentry_upload_proguard         |
|                                   |         | sentry_create_release                             |
| fastlane-plugin-appcenter         | 1.8.0   | appcenter_fetch_version_number                    |
|                                   |         | appcenter_fetch_devices appcenter_upload          |
+-----------------------------------+---------+---------------------------------------------------+

[18:36:01]: ------------------------------------
[18:36:01]: --- Step: ios_lint_localizations ---
[18:36:01]: ------------------------------------
[18:36:01]: Linting localizations for parameter placeholders consistency...
[18:36:01]: $ /Users/olivier/Documents/Dev/simplenote-ios/vendor/swiftgen/6.4.0/bin/swiftgen config run --config /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/swiftgen-stringtypes.yml
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.it.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.es.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.ar.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.el.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.id.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.cy.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.he.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.de.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.en.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.ja.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.pt-BR.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.fr.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.nl.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.fa.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.ru.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.ko.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.sv.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.tr.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.zh-Hans-CN.txt
[18:36:01]: ▸ File written: /var/folders/0y/8j100qcn7vg_3m4l6lhqfzgh0000gn/T/a8c-lint-translations-20201202-21040-o42kre/L10nParamsList.zh-Hant-TW.txt
[18:36:01]: Result: {}
```
</details>

### Review

Only one developer and/or release manager is required to review these changes, but anyone can perform the review.

### Release

These changes do not require release notes.

# PR Requirements before we merge

* This PR requires that https://github.com/wordpress-mobile/release-toolkit/pull/177 gets merged first and that we then create a new version of the release-toolkit to point to it in this PR

### CODE REVIEW

File: fastlane/Pluginfile
Comment (AliSoftware) [inline fastlane/Pluginfile]: ⚠️ This needs to be reverted to point to a tag once wordpress-mobile/release-toolkit#177 is merged and we tag a new version of the release toolkit, before we can merge this PR.

File: .bundle/config
Comment (jkmassel) [inline .bundle/config:2]: `excellent.gif`

**Review by jkmassel — APPROVED** [review]
LGTM once I got past all the localization changes 😅

:shipit: once the tag lands

File: .bundle/config
Comment (mokagio) [reply .bundle/config:2]: Looking forward to Bundler 2.2.0 to be released. It's been in RC for a long time...

The new version fixes the bug where calling `BUNDLE_WITH=group` touches this file (see https://github.com/rubygems/rubygems/issues/3708) which currently makes it confusing to run some of our workflows when this file is tracked because it changes it and people might not know whether to track the change or not.

Internal ref paaHJt-13C

🤞 it won't be long...

File: Simplenote.xcodeproj/project.pbxproj
Comment (AliSoftware) [inline Simplenote.xcodeproj/project.pbxproj:1996]: Updating this for same reason as https://github.com/woocommerce/woocommerce-ios/pull/3275#discussion_r539633982, i.e. since CocoaPods 1.10 now use `compatibilityVersion` to decide if `inputFileList`/`outputFileList` is available, we need that bump here to avoid a bigger diff.

**Comment by AliSoftware** [comment]
@jkmassel @mokagio I've updated the PR:
 - Rebased on top of `release/4.29`, updated the PR's target branch accordingly
 - Pointed the release-toolkit to the recently created new tag 0.11.0
 - Update the behavior for `new_beta_release` to allow bypassing the violations for betas (while still strongly suggesting to abort on violations for beta, and while still making it a non-bypass-able error for `finalize_release`

I'll let you review the new changes; if you're ok with them feel free to merge on green yourself.

File: Simplenote.xcodeproj/project.pbxproj
Comment (mokagio) [reply Simplenote.xcodeproj/project.pbxproj:1996]: 😮 this explains why I got the full list in [this commit on Simplenote macOS with CP 1.10](https://github.com/Automattic/simplenote-macos/pull/743/commits/5456c8bf2d24858ca0f1a94fb0593bc73aba72d2).

Thank you for documenting this 👍

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile:133]: Nice!

Suggestion for future improvement @AliSoftware: move this into the action itself. I'm not sure what could be a good parameter name to enable this behavior 🤔 `interactive: true` maybe?

**Review by mokagio — APPROVED** [review]
Thank you for working on this. Looking forward to see it in action tomorrow when finalizing 4.29 😄

**Pull request merged**
